Brokerages/Backtesting/BacktestingBrokerage.cs

```diff
@@ -191,6 +204,11 @@ public override bool CancelOrder(Order order)
         }
 
         /// <summary>
+        /// Market Simulation - simulates various market conditions in backtest
+        /// </summary>
+        public IBacktestingMarketSimulation MarketSimulation { get; set; }
```

Comment:
The idea is to simulate other market participant's behavior. In our case: option assignment. The algo generates a list of random dates close to expiration when time value of the options is minimal, and on those random dates it tries to find an arb opportunity. if it finds it, it exercises the option and user's algo gets assigned.